<hr style="border-top:5px solid Gold; margin-top: 2px; margin-bottom: 0px">
<hr style="border-top:5px solid Silver; margin-top: 0px; margin-bottom: 0px">
<hr style="border-top:5px solid #DC9233; margin-top: 0px; margin-bottom: 0px">

# Medals at the Olympics

<hr style="border-top:5px solid Gold; margin-top: 2px; margin-bottom: 0px">
<hr style="border-top:5px solid Silver; margin-top: 0px; margin-bottom: 0px">
<hr style="border-top:5px solid #DC9233; margin-top: 0px; margin-bottom: 0px">

In [1]:
# Standard imports
import pandas as pd

# Visulation Imports
import matplotlib.pyplot as plt
import seaborn as sns

# import splitting and imputing functions
from sklearn.model_selection import train_test_split
from sklearn.impute import SimpleImputer

# Modeling imports
from sklearn.linear_model import LinearRegression, LassoLars, TweedieRegressor
from sklearn.metrics import mean_squared_error, r2_score, explained_variance_score
from sklearn.feature_selection import f_regression 
from sklearn.metrics import mean_squared_error
from sklearn.preprocessing import PolynomialFeatures
from sklearn.feature_selection import SelectKBest, f_regression, RFE
import sklearn.preprocessing
import statsmodels.api as sm
from sklearn.cluster import DBSCAN
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import classification_report, accuracy_score
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.naive_bayes import MultinomialNB
from sklearn.tree import DecisionTreeClassifier
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import cross_val_score

# turn off pink boxes
import warnings
warnings.filterwarnings("ignore")

# Custom Imports
import wrangle
import explore

## Project Goal
- Create a classification model to predict if an olympic athlete will medal or not.

<hr style="border-top:5px solid Gold; margin-top: 2px; margin-bottom: 0px">
<hr style="border-top:5px solid Silver; margin-top: 0px; margin-bottom: 0px">
<hr style="border-top:5px solid #DC9233; margin-top: 0px; margin-bottom: 0px">

# Acquire and Prepare

<hr style="border-top:5px solid Gold; margin-top: 2px; margin-bottom: 0px">
<hr style="border-top:5px solid Silver; margin-top: 0px; margin-bottom: 0px">
<hr style="border-top:5px solid #DC9233; margin-top: 0px; margin-bottom: 0px">

#### This function reads 2 csv's and assigns them into a dataframe. It then feature engineers 5 new columns, those columns are Medal, Sex_Male, medalist, BMI, and AgeBins. From there it splits the data and imputes the null values for Age, drops data prior to the year 1961, and drops the remaining nulls.

In [2]:
df, df1, train, validate, test = wrangle.wrangle_df()

ValueError: Found input variables with inconsistent numbers of samples: [216892, 271116]

In [ ]:
train.head()

## Acquire and Prepare Takeaways

- 5 columns were feature engineered.
- Nulls were handled by imputing with mean, dropping old data, and dropping the remaining nulls.
- Our dataframe was left with 271,116 rows and 19 columns.

<hr style="border-top:5px solid Gold; margin-top: 2px; margin-bottom: 0px">
<hr style="border-top:5px solid Silver; margin-top: 0px; margin-bottom: 0px">
<hr style="border-top:5px solid #DC9233; margin-top: 0px; margin-bottom: 0px">

# Exploration

<hr style="border-top:5px solid Gold; margin-top: 2px; margin-bottom: 0px">
<hr style="border-top:5px solid Silver; margin-top: 0px; margin-bottom: 0px">
<hr style="border-top:5px solid #DC9233; margin-top: 0px; margin-bottom: 0px">

## Lets look at the correlations in the data

In [ ]:
explore.correlation_heatmap(train)

## Correlation Heatmap Takeaways
- There are no negative correlation in this dataset.
- There does not appear to be any strong correlations with the current features.

In [ ]:
groups = train.groupby('Team').sum()
groups = groups[groups.medalist > 500].reset_index()
groups

In [ ]:
# plt.figure(figsize=(10, 10))
# sns.barplot(x='Team', y='medalist', data=train)